### PR TITLE
persist bot tokens in S3

### PR DESCRIPTION
**Pull request by martinklepsch**
[Trello card](https://trello.com/c/QEAeBkKE/428-org-bot-id-storage-in-auth-include-in-jwtokens) — depends on https://github.com/belucid/open-company-infrastructure/pull/18

This adds a small layer of persistence to the Auth service to store secrets we need might need to trigger bot conversations. It stores these secrets in S3 for now. As needs evolve we can re-evaluate other options. (Previous discussions suggested Redis & RethinkDB, I'm now thinking DynamoDB via [faraday-atom](https://github.com/danstone/faraday-atom/) might also be a good option.)

The `bot-integration` [PR to API](https://github.com/open-company/open-company-api/pull/43) has been updated to use the presence of bot tokens as requirement instead of checking that the user is an owner of the team: https://github.com/open-company/open-company-api/pull/43/commits/a513a8e2cd98d50eeec9bbf85d16f3719af44929 

The token-writing is synchronous so authentication may take a few hundred ms extra. This should only be the case if a bot token is present and it's different to the one already stored in the database so a minor portion auth-flows. 

**To Test:**
- sign out
- sign in via Slack: **make sure the scopes include `bot`**
- check the `store` file in the S3 bucket
  - it should contain your Slack org-id
  - and the bot-id and the bot-secret

You can also try the same with the orgs/users provided in the infrastructure repo. Once an owner has signed in all tokens issued afterwards should contain the bot secrets (and these should be stored in the `store`/`store-production` files in the S3 bucket).

### CODE REVIEW

**Comment by belucid** [comment]
Reviewing.

File: src/open_company_auth/app.clj [hunk 1]
```diff
@@ -49,35 +52,26 @@
...
-      "Running on port: " port "\n"
-      "Hot-reload: " config/hot-reload "\n"
-      "Sentry: " config/dsn "\n\n"
-      "Ready to serve...\n")))
+  (println (str "\n" (slurp (io/resource "ascii_art.txt")) "\n"
+                "OpenCompany Auth Server\n"
+                "Running on port: " port "\n"
+                "Hot-reload: " config/hot-reload "\n"
+                "Sentry: " config/dsn "\n\n"
+                "Ready to serve...\n")))
```
Comment (belucid) [inline src/open_company_auth/app.clj:76]: It'd be good @martinklepsch to add the S3 bucket being used to the startup output.

Comment (martinklepsch) [inline src/open_company_auth/app.clj:76]: Currently the bucket is always the same and we're just using different files for dev/prod. I will add a line indicating the used `bucket/key`

**Comment by belucid** [comment]
@martinklepsch should document the AWS env settings (access/secret) needed for the auth to run in the README.

**Comment by martinklepsch** [comment]
Yes, good idea, I'll add the Slack stuff as well, I remember this has been confusing me when I first tried to run `auth`.

**Comment by belucid** [comment]
@martinklepsch cool

**Comment by martinklepsch** [comment]
@belucid updated

**Comment by belucid** [comment]
Reviewing.

**Comment by belucid** [comment]
Testing.

File: README.md [hunk 2]
```diff
@@ -67,15 +67,17 @@ Make sure you update the section in `project.clj` that looks like this to contai
 :dev [:qa {
   :env ^:replace {
     :open-company-auth-passphrase "this_is_a_dev_secret" ; JWT secret
-    :hot-reload "true" ; reload code when changed on the file system
     :open-company-slack-client-id "FIXME"
     :open-company-slack-client-secret "FIXME"
-    :aws-access-key "FIXME"
-    :aws-secret-key "FIXME"
```
Comment (martinklepsch) [inline README.md]: These are the words used by AWS `access-key` & `secret-key`, I don't see the point of changing them here?

**Pull request merged**